MacawTests/SceneSerialization.swift

```diff
@@ -0,0 +1,529 @@
+            let color = Color(dictionary: fillDict) {
+            fill = color
+        }
+        if let strokeDict = dictionary["stroke"] as? [String:Any] {
+            stroke = Stroke(dictionary: strokeDict)
+        }
+        if let alignString = dictionary["align"] as? String {
+            align = Align.instantiate(string: alignString)
+        }
+        if let baselineString = dictionary["baseline"] as? String {
+            baseline = baselineForString(baselineString)
+        }
+        
+        fromDictionary(dictionary: dictionary) // fill in the fields inherited from Node
+    }
+    
+    fileprivate func baselineForString(_ string: String) -> Baseline {
+        switch(string) {
+        case "top": return .top
+        case "alphabetic": return .alphabetic
+        case "bottom": return .bottom
+        case "mid": return .mid
+        default: return .top
+        }
+    }
+}
+
+extension Group: Serializable {
+    
+    func toDictionary() -> [String:Any] {
+        var nodes = [[String:Any]]()
+        for node in contents {
+            if let node = node as? Serializable {
+                nodes.append(node.toDictionary())
+            }
+        }
+        var result = super.baseToDictionary()
+        result["node"] = "Group"
+        result["contents"] = nodes
+        return result
+    }
+    
+    convenience init?(dictionary: [String:Any]) {
+        
+        guard let contents = dictionary["contents"] as? [[String:Any]] else {
+            return nil
+        }
+        var nodes = [Node]()
+        for dict in contents {
+            if let node = Node.instantiate(dictionary: dict) {
+                nodes.append(node)
+            }
+        }
+        self.init()
+        self.contents = nodes
+        fromDictionary(dictionary: dictionary) // fill in the fields inherited from Node
+    }
+}
+
+
+
+extension Locus {
+    static func instantiate(dictionary: [String:Any]) -> Locus? {
+        guard let type = dictionary["type"] as? String else {
+            fatalError("No type specified")
+        }
+        if type == "Arc" {
+            return Arc(dictionary: dictionary)
+        }
+        if type == "Circle" {
+            return Circle(dictionary: dictionary)
+        }
+        if type == "Ellipse" {
+            return Ellipse(dictionary: dictionary)
+        }
+        if type == "Line" {
+            return Line(dictionary: dictionary)
+        }
+        if type == "Path" {
+            return Path(dictionary: dictionary)
+        }
+        if type == "Polygon" {
+            return Polygon(dictionary: dictionary)
+        }
+        if type == "Polyline" {
+            return Polyline(dictionary: dictionary)
+        }
+        if type == "Rect" {
+            return Rect(dictionary: dictionary)
+        }
+        if type == "RoundRect" {
+            return RoundRect(dictionary: dictionary)
+        }
+        
+        print("Locus from dictionary error. Locus \(type) not supported")
```

Comment:
Usually it's easier to just fail there and avoid all these null values handling.

Reply:
null values? These are just unsupported or misspelled node names, or did I not understand you correctly?

Reply:
There is `return null` after this line. If you just throw fatal error you don't need to handle optional value then.